src/elfpy/agent.py

```diff
@@ -1,142 +1,70 @@
 
 from elfpy.utils.outputs import float_to_string
 from elfpy.utils.bcolors import Bcolors as bcolors
+from elfpy.wallet import Wallet
 
 
-@dataclass(frozen=False)
-class AgentWallet:
-    """Stores what's in the agent's wallet"""
-
-    # TODO: Create our own dataclass decorator that is always mutable and includes dict set/get syntax
-    # pylint: disable=duplicate-code
-
-    # fungible
-    base_in_wallet: float
-    lp_in_wallet: float = 0  # they're fungible!
-    fees_paid: float = 0
-    # non-fungible (identified by mint_time, stored as dict)
-    token_in_wallet: dict = field(default_factory=dict)
-    base_in_protocol: dict = field(default_factory=dict)
-    token_in_protocol: dict = field(default_factory=dict)
-    effective_price: float = field(init=False)  # calculated after init, only for transactions
-
-    def __post_init__(self):
-        """Post initialization function"""
-        # check if this represents a trade (one side will be negative)
-        total_tokens = sum(list(self.token_in_wallet.values()))
-        if self.base_in_wallet < 0 or total_tokens < 0:
-            self.effective_price = total_tokens / self.base_in_wallet
-
-    def __getitem__(self, key):
-        return getattr(self, key)
-
-    def __setitem__(self, key, value):
-        setattr(self, key, value)
-
-    def __str__(self):
-        output_string = ""
-        for key, value in vars(self).items():
-            if value:  #  check if object exists
-                if value != 0:
-                    output_string += f" {key}: "
-                    if isinstance(value, float):
-                        output_string += f"{float_to_string(value)}"
-                    elif isinstance(value, list):
-                        output_string += "[" + ", ".join([float_to_string(x) for x in value]) + "]"
-                    elif isinstance(value, dict):
-                        output_string += "{" + ", ".join([f"{k}: {float_to_string(v)}" for k, v in value.items()]) + "}"
-                    else:
-                        output_string += f"{value}"
-        return output_string
-
-
-# TODO: The user class has too many instance attributes (8/7)
+# TODO: The agent class has too many instance attributes (8/7)
 #     we should move some, like budget and wallet_address, into the agent wallet and out of User
-# pylint: disable=too-many-instance-attributes
-class User:
+class Agent:
     """
-    Implements abstract classes that control user behavior
-    user has a budget that is a dict, keyed with a date
+    Implements abstract classes that control agent behavior
+    agent has a budget that is a dict, keyed with a date
     value is an inte with how many tokens they have for that date
     """
 
-    # TODO: variables assigned by child classes are referenced by User -- need to fix up user inheritance
+    # TODO: variables assigned by child classes are referenced by User -- need to fix up agent inheritance
```

Comment:
User -> Agent
Although I think this PR might be addressing this TODO?

Reply:
yes it did!